src/mod/applications/mod_sms/mod_sms.c

```diff
@@ -514,6 +514,13 @@ SWITCH_STANDARD_CHAT_APP(stop_function)
 	return SWITCH_STATUS_FALSE;
 }
 
+SWITCH_STANDARD_CHAT_APP(final_function)
+{
+	switch_event_add_header(message, SWITCH_STACK_BOTTOM, "final_delivery", "true");
+	switch_set_flag(message, EF_NO_CHAT_EXEC);
+	return SWITCH_STATUS_FALSE;
```

Comment:
This looks wrong and the one above looks wrong too. Shouldn't them return success?

Reply:
returning false stops processing in mod_sms.  I think this is correct.